enumeration.rb

```diff
@@ -0,0 +1,97 @@
+module Enumerable
+  def my_each
+    # Should return the array except block is given
+    return to_enum unless block_given?
+
+    each do |i|
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    # Should return the array except block is given
+    return to_enum unless block_given?
+
+    (0..size - 1).my_each do |i|
+      yield(self[i], i)
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    # an empty array to store our selected element
+    arr = []
+    (0...size - 1).my_each do |i|
+      arr << self[i] if yield(self[i])
+    end
+    arr
+  end
+
+  def my_all?
+    return true unless block_given?
+
+    count = 0
+    size.times do |i|
+      count += 1 unless yield self[i]
+    end
+    count.zero?
+  end
+
+  def my_any?
+    return true unless block_given?
+
+    count = 0
+    size.times do |i|
+      count += 1 unless yield self[i]
+    end
+    count != 0
+  end
+
+  def my_none?
+    return false unless block_given?
```

Comment:
It doesn't return false for all cases where no block is given. If no block is given, and all the elements are false values, my_none? should return true.